docs/dev-guide/plugin-dev.md

```diff
@@ -4,199 +4,331 @@ sidebarDepth: 3
 
 # Plugin Development Guide
 
-## Core Concepts
+## Getting started
 
-There are two major parts of the system:
+A CLI plugin is an npm package that can add additional features to a `@vue/cli` project. It should always contain a [Service Plugin](#service-plugin) as its main export, and can optionally contain a Generator, a Prompt File and a Vue UI integration.
```

Comment:
> a `@vue/cli` project

What would you think about "a project using Vue CLI" instead? My thinking is to reference "Vue CLI" as an ecosystem of projects rather than `@vue/cli` as a package, since `@vue/cli` may not have actually been used to create the project in cases where someone upgraded or used a boilerplate - plus, `@vue/cli` will not be listed as a dependency in the project itself, which may cause confusion.

Specifying "a project using Vue CLI" instead of "a Vue CLI project" also differentiates between plugins that _tie into_ Vue CLI and applications/libraries that _make use of_ the Vue CLI ecosystem.

> It should always contain a [Service Plugin](#service-plugin) as its main export, and can optionally contain a Generator, a Prompt File and a Vue UI integration.

At a high level, what would you think about starting with user goals rather than implementation details? We're getting into the "how" of a plugin when we still haven't really answered the "why". As a user, I might be wondering, "How do I know I want to write a plugin? What are plugins for?" This paragraph contains _hints_ that I can receive options from the user, add/update project files, and add to the Vue CLI UI, but that's only the tip of the iceberg. Plus, it doesn't give me an idea of what a plugin _isn't_ appropriate for. For example, Vuetify has a plugin (because they want to make some changes to the Webpack config). Does this mean all component libraries need a plugin? Would I create a plugin to include lodash in a project?

> Generator, a Prompt File, and a Vue UI integration

Lists like this are a great place for links, for users who want to jump directly to one of these topics. If you're not sure what you want to yet or you don't want to break your flow while writing, I often just make the link `TODO` so I don't forget to come back to it (e.g. `[Generator](TODO)`).